lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -136,114 +138,136 @@ void FirRegLower::lower() {
-
-                  if (asyncReset) {
-                    auto &[resetSignal, resetValue] = *asyncReset;
-                    resets[resetSignal].emplace_back(svReg, resetValue);
-                  }
-                }
-              });
+              // Create initialisers for all registers.
+              for (auto &svReg : toInit) {
+                initialize(builder, svReg, randValues);
 
-              if (!resets.empty()) {
-                builder.create<sv::IfDefProceduralOp>("RANDOMIZE", [&] {
-                  // If the register is async reset, we need to insert extra
-                  // initialization in post-randomization so that we can set the
-                  // reset value to register if the reset signal is enabled.
-                  for (auto &reset : resets) {
-                    // Create a block guarded by the RANDOMIZE macro and the
-                    // reset: `ifdef RANDOMIZE
-                    //   if (reset) begin
-                    //     ..
-                    //   end
-                    // `endif
-                    builder.create<sv::IfOp>(reset.first, [&] {
-                      for (auto &[reg, value] : reset.second)
-                        builder.create<sv::BPAssignOp>(reg.getLoc(), reg,
-                                                       value);
-                    });
-                  }
-                });
+                if (svReg.asyncResetSignal)
+                  resets[svReg.asyncResetSignal].emplace_back(svReg);
               }
             });
 
-            builder.create<sv::IfDefOp>("FIRRTL_AFTER_INITIAL", [&] {
-              builder.create<sv::VerbatimOp>("`FIRRTL_AFTER_INITIAL");
-            });
+            if (!resets.empty()) {
+              builder.create<sv::IfDefProceduralOp>("RANDOMIZE", [&] {
+                // If the register is async reset, we need to insert extra
+                // initialization in post-randomization so that we can set the
+                // reset value to register if the reset signal is enabled.
+                for (auto &reset : resets) {
+                  // Create a block guarded by the RANDOMIZE macro and the
+                  // reset: `ifdef RANDOMIZE
+                  //   if (reset) begin
+                  //     ..
+                  //   end
+                  // `endif
+                  builder.create<sv::IfOp>(reset.first, [&] {
+                    for (auto &reg : reset.second)
+                      builder.create<sv::BPAssignOp>(reg.reg.getLoc(), reg.reg,
+                                                     reg.asyncResetValue);
+                  });
+                }
+              });
+            }
+          });
+
+          builder.create<sv::IfDefOp>("FIRRTL_AFTER_INITIAL", [&] {
+            builder.create<sv::VerbatimOp>("`FIRRTL_AFTER_INITIAL");
           });
         });
-  }
+      });
 
   module->removeAttr("firrtl.random_init_width");
```

Comment:
Note to self/anyone: Can we have our dialect attribute strings in a common location? :innocent:

Reply:
It should at least be on the dialect ODS so it goes on the website 😬 . In general, I'm not sure best practice for dialect attrs. Maybe put the string into a constant somewhere?